volatility3/framework/plugins/windows/skeleton_key_check.py

```diff
@@ -0,0 +1,578 @@
+                                                                    interfaces.configuration.path_join(self.config_path, 'cryptdll'),
+                                                                    proc_layer_name,
+                                                                    "cryptdll.pdb",
+                                                                    cryptdll_base,
+                                                                    cryptdll_size)
+        except exceptions.VolatilityException:
+            return None, None, None
+
+        array_start, count, rc4HmacInitialize, rc4HmacDecrypt = \
+            self._find_array_with_pdb_symbols(cryptdll_symbols, cryptdll_types, proc_layer_name, cryptdll_base) 
+       
+        try: 
+            array = cryptdll_types.object(object_type = "array",
+                                    offset = array_start,
+                                    subtype = cryptdll_types.get_type("_KERB_ECRYPT"),
+                                    count = count,
+                                    absolute = True)
+
+        except exceptions.InvalidAddressException:
+            vollog.debug("The CSystem array is not present in memory. Stopping PDB symbols based analysis.")
+            return None, None, None
+
+        return array, rc4HmacInitialize, rc4HmacDecrypt
+
+    def _get_rip_relative_target(self, inst) -> int:
+        """
+        Returns the target address of a RIP-relative instruction.
+
+        These instructions contain the offset of a target address
+        relative to the current instruction pointer.
+        
+        Args:
+            inst: A capstone instruction instance
+
+        Returns:
+            None or the target address of the instruction
+        """
+        try:
+            opnd = inst.operands[1]
+        except capstone.CsError: 
+            return None
+
+        if opnd.type != capstone.x86.X86_OP_MEM:
+            return None
+
+        if inst.reg_name(opnd.mem.base) != "rip":
+            return None
+
+        return inst.address + inst.size + opnd.mem.disp
+
+    def _analyze_cdlocatecsystem(self, function_bytes: bytes,
+                                       function_start: int, 
+                                       proc_layer_name: str) -> Tuple[int, int]:
+        """
+        Performs static analysis on CDLocateCSystem to find the instructions that
+        reference CSystems as well as cCsystems
+
+        Args:
+            function_bytes: the instruction bytes of CDLocateCSystem
+            function_start: the address of CDLocateCSystem
+            proc_layer_name: the name of the lsass.exe process layer
+
+        Return:
+            Tuple of:
+            array_start: address of CSystem
+            count: the count from cCsystems or 16
+        """
+        found_count = False
+        array_start = None
+        count = None
+
+        ## we only support 64bit disassembly analysis
```

Comment:
We check that the sample is 64bit. The real lsass won't be 32bit on a 64 bit system and the 64bit lsass will load the 64bit cryptdll, so I think we are okay.

Reply:
I guess my question is, where do we ensure the sample is 64-bit?  The `primary` requirement seems to allow for `intel32` architecture, so that was why I was asking if a check was needed.  I might've missed it, but I couldn't see anything from the generator down that checked it was explicitly a 64-bit image?

Reply:
the check and message is here: https://github.com/volatilityfoundation/volatility3/blob/68a6fd252f686f386d60565281cf93655c2c98c6/volatility3/framework/plugins/windows/skeleton_key_check.py#L500

Reply:
Oh, ok, I was just being blind then, cool!  5:D